packages/nextjs/src/server/getAuth.ts

```diff
@@ -10,42 +10,65 @@ import {
 } from '@clerk/backend';
 import type { SecretKeyOrApiKey } from '@clerk/types';
 
+import { createDebugLogger, withLogger } from '../utils/debugLogger';
 import { API_KEY, API_URL, API_VERSION, SECRET_KEY } from './clerkClient';
+import { getAuthAuthHeaderMissing } from './errors';
 import type { RequestLike } from './types';
 import { getAuthKeyFromRequest, getCookie, getHeader, injectSSRStateIntoObject } from './utils';
 
 type GetAuthOpts = Partial<SecretKeyOrApiKey>;
-export const getAuth = (req: RequestLike, opts?: GetAuthOpts): SignedInAuthObject | SignedOutAuthObject => {
-  // When the auth status is set, we trust that the middleware has already run
-  // Then, we don't have to re-verify the JWT here,
-  // we can just strip out the claims manually.
-  const authStatus = getAuthKeyFromRequest(req, 'AuthStatus');
-  const authMessage = getAuthKeyFromRequest(req, 'AuthMessage');
-  const authReason = getAuthKeyFromRequest(req, 'AuthReason');
 
-  if (!authStatus) {
-    throw new Error(
-      'You need to use "withClerkMiddleware" in your Next.js middleware file. You also need to make sure that your middleware matcher is configured correctly and matches this route or page. See https://clerk.com/docs/quickstarts/get-started-with-nextjs',
-    );
-  }
+export const createGetAuth = ({
+  debugLoggerName,
+  noAuthStatusMessage,
+}: {
+  noAuthStatusMessage: string;
+  debugLoggerName: string;
+}) =>
+  withLogger(createDebugLogger(debugLoggerName), logger => {
+    return (req: RequestLike, opts?: GetAuthOpts): SignedInAuthObject | SignedOutAuthObject => {
+      const debug = getHeader(req, constants.Headers.EnableDebug) === 'true';
+      if (debug) {
+        logger.enable();
+      }
 
-  const options = {
-    apiKey: opts?.apiKey || API_KEY,
-    secretKey: opts?.secretKey || SECRET_KEY,
-    apiUrl: API_URL,
-    apiVersion: API_VERSION,
-    authStatus,
-    authMessage,
-    authReason,
-  };
+      // When the auth status is set, we trust that the middleware has already run
+      // Then, we don't have to re-verify the JWT here,
+      // we can just strip out the claims manually.
+      const authStatus = getAuthKeyFromRequest(req, 'AuthStatus');
+      const authMessage = getAuthKeyFromRequest(req, 'AuthMessage');
+      const authReason = getAuthKeyFromRequest(req, 'AuthReason');
+      logger.log('Headers debug', { authStatus, authMessage, authReason });
 
-  if (authStatus !== AuthStatus.SignedIn) {
-    return signedOutAuthObject(options);
-  }
+      if (!authStatus) {
+        throw new Error(noAuthStatusMessage);
+      }
```

Comment:
For the reviewers: This has not changed, this is just terrible git diff formatting